docs/_reference/types.md

```diff
@@ -0,0 +1,1369 @@
+---
+title: "@slack/types"
+slug: types
+---
+
+<h1 id="slacktypes">@slack/types</h1>
+<h2 id="interfaces">Interfaces</h2>
+<h3 id="action">Action</h3>
+<h4 id="fields">Fields</h4>
+<table>
+<thead>
+<tr>
+<th align="center">Name</th>
+<th align="center">Type</th>
+<th></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td align="center">action_id</td>
+<td align="center"><code>string</code></td>
+<td></td>
+</tr>
+<tr>
+<td align="center">type</td>
+<td align="center"><code>string</code></td>
+<td></td>
+</tr>
+</tbody>
+</table>
+<h3 id="actionsblock">ActionsBlock</h3>
+<h4 id="fields-1">Fields</h4>
+<table>
+<thead>
+<tr>
+<th align="center">Name</th>
+<th align="center">Type</th>
+<th></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td align="center">elements</td>
+<td align="center"><code>(<a href="#button" title="">Button</a> | <a href="#overflow" title="">Overflow</a> | <a href="#datepicker" title="">Datepicker</a> | <a href="#select" title="">Select</a> | <a href="#radiobuttons" title="">RadioButtons</a> | <a href="#action" title="">Action</a>)[]</code></td>
+<td></td>
+</tr>
+<tr>
+<td align="center">type</td>
+<td align="center"><code>'actions'</code></td>
+<td></td>
+</tr>
+</tbody>
+</table>
+<h3 id="attachmentaction">AttachmentAction</h3>
+<h4 id="fields-2">Fields</h4>
+<table>
+<thead>
+<tr>
+<th align="center">Name</th>
+<th align="center">Type</th>
+<th></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td align="center">confirm</td>
+<td align="center"><code><a href="#confirmation" title="">Confirmation</a></code></td>
+<td></td>
+</tr>
+<tr>
+<td align="center">data_source</td>
+<td align="center"><code>'static' | 'channels' | 'conversations' | 'users' | 'external'</code></td>
```

Comment:
A TODO for the future: types in tables consisting of unions (and also intersections!) could use some cleaning. 🧹